File: languages/python/bloock/entity/identity_v2/boolean_attribute.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_v2_pb2 as proto
from bloock.entity.identity_v2.attribute import Attribute


class BooleanAttribute(Attribute):
    @staticmethod
    def from_proto(a: proto.BooleanAttributeV2) -> BooleanAttribute:
        return BooleanAttribute(
            id=a.id,
            value=a.value,
        )

    def to_proto(self) -> proto.BooleanAttributeV2:
        return proto.BooleanAttributeV2(
            id=self.id,
            value=self.value,
        )

File: languages/js/src/entity/identity_v2/boolean_attribute.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities_v2";
import { Attribute } from "./attribute";

/**
 * Represents an attribute with a boolean value.
 */
export class BooleanAttribute extends Attribute<boolean> {
  public toProto(): identityEntitiesProto.BooleanAttributeV2 {
    return identityEntitiesProto.BooleanAttributeV2.fromPartial({
      id: this.id,
      value: this.value
    });
  }

  static fromProto(
    r: identityEntitiesProto.BooleanAttributeV2
  ): BooleanAttribute {
    return new BooleanAttribute(r.id, r.value);
  }
}
